kroxylicious/src/main/java/io/kroxylicious/proxy/config/FilterDefinition.java

```diff
@@ -9,7 +9,7 @@
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver;
 
-import io.kroxylicious.proxy.internal.filter.FilterConfig;
+import io.kroxylicious.proxy.filter.FilterConfig;
```

Comment:
Hmm. That's interesting we are promoting `FilterConfig` to a first class part of the API. Do we actually need to do that? Does FilterConfig actually give us something generically useful? 

Is there a better way of configuring filters that reduces our API surface?

Reply:
Good question, I promoted it because the existing filter implementations extend it when defining their config classes, so it's part of the current filter api. It doesn't have any of it's own functionality yet but I'd say it's nicer to read the FilterContributors when they are loading `<? extends FilterConfig>` rather than just some `Object`.

I might have an explore to see if we could use kafka's `ConfigDef`, I imagine this would be more verbose but we wouldn't have to document how to build config class and annotate it correctly using jackson annotation.